src/Integration.UnitTests/Binding/ProjectBinderFactoryTests.cs

```diff
@@ -21,21 +21,33 @@
 using System;
 using System.IO.Abstractions;
 using FluentAssertions;
+using Microsoft.VisualStudio.ComponentModelHost;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 using Moq;
+using SonarLint.VisualStudio.Core;
 using SonarLint.VisualStudio.Integration.Binding;
 
 namespace SonarLint.VisualStudio.Integration.UnitTests.Binding
 {
     [TestClass]
     public class ProjectBinderFactoryTests
     {
+        private Mock<IProjectToLanguageMapper> projectToLanguageMapper;
         private ProjectBinderFactory testSubject;
 
         [TestInitialize]
         public void TestInitialize()
         {
-            testSubject = new ProjectBinderFactory(Mock.Of<IServiceProvider>(), Mock.Of<ILogger>(), Mock.Of<IFileSystem>());
+            projectToLanguageMapper = new Mock<IProjectToLanguageMapper>();
+
+            var serviceProvider = new ConfigurableServiceProvider();
```

Comment:
Previously we would pass a Mock serviceProvider and all the GetService would simply return null.
To support GetMefService<> I had to use ConfigurableComponentModel and ConfigurableServiceProvider, as GetMefService uses extension methods. This unfortunately then lead to a bunch of errors that all these dependencies could not resolved, even though they're not really used.
Hence I'm registering them with mock values.